lib/transmuxer/mpeg_audio.js

```diff
@@ -0,0 +1,190 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.transmuxer.MpegAudio');
+
+
+/**
+ *  MPEG parser utils
+ */
+shaka.transmuxer.MpegAudio = class {
+  /**
+   * @param {!Uint8Array} data
+   * @param {!number} offset
+   * @return {?{sampleRate: number, channelCount: number,
+   * frameLength: number, samplesPerFrame: number}}
+   */
+  static parseHeader(data, offset) {
+    const MpegAudio = shaka.transmuxer.MpegAudio;
+
+    const mpegVersion = (data[offset + 1] >> 3) & 3;
```

Comment:
I always find it useful to have some comments here with links to reference material used to create these parsers.  It is very helpful when debugging or verifying them.

Reply:
Wikipedia has a good diagram about the file structure https://en.wikipedia.org/wiki/MP3 . I added a comment in the class